### PR TITLE
Refactor and document for student use

### PR DESCRIPTION
**Pull request by nrc**
This PR is organised into self-contained commits. It uses the TestXxx4A format to name tests, uses the `Server` interface rather than the `Command` interface. I have marked a bunch of code for deletion in the final version, and documented the proto defs (which included deleting some dead code and moving some messages around).

PTAL @Connor1996

### CODE REVIEW

File: proto/proto/kvrpcpb.proto [hunk 1]
```diff
@@ -155,88 +168,94 @@ message CheckTxnStatusRequest {
...
 
-message RawGetResponse {
-    errorpb.Error region_error = 1;
-    string error = 2;
-    bytes value = 3;
-    bool not_found = 4;
+enum Op {
+    Put = 0;
+    Del = 1;
+    Rollback = 2;
```
Comment (Connor1996) [inline proto/proto/kvrpcpb.proto]: can you help me add a type `Lock = 3; // used by TinySQL to add index, though it's won't implemented by TinyKV`

File: kv/tikv/transaction/commands4a_test.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+import (
+	"testing"
+
+	"github.com/pingcap-incubator/tinykv/kv/util/engine_util"
+	"github.com/pingcap-incubator/tinykv/proto/pkg/kvrpcpb"
+	"github.com/stretchr/testify/assert"
+)
+
+// TestGetValue4A getting a value works in the simple case.
+func TestGetValue4A(t *testing.T) {
```
Comment (Connor1996) [inline kv/tikv/transaction/commands4a_test.go]: why these are empty?

File: proto/proto/kvrpcpb.proto (hunk 1 above)
Comment (nrc) [reply proto/proto/kvrpcpb.proto]: Sure!

File: kv/tikv/transaction/commands4a_test.go (hunk 2 above)
Comment (nrc) [reply kv/tikv/transaction/commands4a_test.go]: I thought we wanted the students to fill these out? I can fill them out with the actual tests if we want.

Comment (Connor1996) [reply kv/tikv/transaction/commands4a_test.go]: students don't need to fill the test, all tests are provided by us to make sure their implementation is correct.

**Pull request merged**